apps/desktop/src/lib/commit/CommitMessageInput.svelte

```diff
@@ -68,10 +71,30 @@
 		return `${title}\n\n${description}`;
 	}
 
-	async function generateCommitMessage(files: LocalFile[]) {
-		const hunks = files.flatMap((f) =>
-			f.hunks.filter((h) => $selectedOwnership.isSelected(f.id, h.id))
-		);
+	async function generateCommitMessage() {
+		let diffInput: DiffInput[] | undefined = undefined;
+
+		if (existingCommit) {
```

Comment:
Little nit, but what about somethign like this instead maybe?

```ts
	const files = await listCommitFiles(project.id, existingCommit.id);
	const diffInput = existingCommit
	 ? files.flatMap((file) =>
			file.hunks.map((hunk) => ({
				filePath: file.path,
				diff: hunk.diff
			}))
		)
	: $branch.files.flatMap((f) =>
			f.hunks
				.filter((h) => $selectedOwnership.isSelected(f.id, h.id))
				.map((h) => ({
					filePath: f.path,
					diff: h.diff
				}))
		);
```

Reply:
The issue there is that I want to call the `listCommitFiles` API only if the `existingCommit` is present.

How about we split the difference?
```ts
	async function getDiffInput(): Promise<DiffInput[]> {
		if (!existingCommit) {
			return $branch.files.flatMap((f) =>
				f.hunks
					.filter((h) => $selectedOwnership.isSelected(f.id, h.id))
					.map((h) => ({
						filePath: f.path,
						diff: h.diff
					}))
			);
		}

		const files = await listCommitFiles(project.id, existingCommit.id);
		return files.flatMap((file) =>
			file.hunks.map((hunk) => ({
				filePath: file.path,
				diff: hunk.diff
			}))
		);
	}

	async function generateCommitMessage() {
		const diffInput = await getDiffInput();
         //...
```